packages/tools/pluggable-widgets-tools/scripts/e2e.js

```diff
@@ -49,51 +60,117 @@ async function main() {
         );
     }
 
-    // Build testProject via mxbuild
     const projectFile = ls("tests/testProject/*.mpr").toString();
+
+    const nativePlatforms = process.env.TRAVIS === "true" ? "=android" : "";
     execSync(
-        `docker run -t -v ${process.cwd()}:/source ` +
-            `--rm mxbuild:${latestMendixVersion} -o /tmp/automation.mda --loose-version-check /source/${projectFile}`,
+        `docker run -t -v ${process.cwd()}:/source --rm mxbuild:${latestMendixVersion} bash -c "mx update-widgets --loose-version-check /source/${projectFile} && mxbuild ` +
+            (isNativeEnabled
+                ? `--native-packager --native-packager-platform${nativePlatforms} --disable-native-animations `
+                : "") +
+            `-o /tmp/automation.mda /source/${projectFile}"`,
         { stdio: "inherit" }
     );
+    console.log("Bundle created and all the widgets are updated");
 
-    // Spin up the runtime and run testProject
-    const freePort = await findFreePort(3000);
-    const runtimeContainerId = execSync(
-        `docker run -td -v ${process.cwd()}:/source -v ${__dirname}:/shared:ro -w /source -p ${freePort}:8080 ` +
-            `-u root -e MENDIX_VERSION=${latestMendixVersion} --entrypoint /bin/bash ` +
-            `mendix/runtime-base:${latestMendixVersion}-rhel /shared/runtime.sh`
-    ).toString();
+    let nativeApp;
+    let runtimeContainerId;
+    try {
+        // Spin up the runtime and run testProject
+        const runtimePort = await findFreePort(3000);
+        runtimeContainerId = execSync(
+            `docker run -td -v ${process.cwd()}:/source -v ${__dirname}:/shared:ro -w /source -p ${runtimePort}:8080 ` +
+                `-u root -e MENDIX_VERSION=${latestMendixVersion} --entrypoint /bin/bash ` +
+                `mendix/runtime-base:${latestMendixVersion}-rhel /shared/runtime.sh`
+        ).toString();
+        await waitForAvailability(`http://localhost:${runtimePort}`, "runtime");
 
-    let attempts = 60;
-    for (; attempts > 0; --attempts) {
-        try {
-            const response = await fetch(`http://localhost:${freePort}`);
-            if (response.ok) {
-                break;
-            }
-        } catch (e) {
-            console.log(`Could not reach http://localhost:${freePort}, trying again...`);
-        }
-        await new Promise(resolve => setTimeout(resolve, 3000));
-    }
+        if (isNativeEnabled) {
+            const nativeAppPaths = await downloadNativeApps();
 
-    try {
-        if (attempts === 0) {
-            throw new Error("Runtime didn't start in time, existing now...");
+            // Android starts
```

Comment:
Maybe extract adnroid/ios setup to functions to make this code a bit more readable?